great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -26,4 +37,76 @@ class SnowflakeDatasource(SQLDatasource):
+    warehouse: Optional[str] = None
+    role: Optional[str] = None
+    numpy: bool = False
+
+    @pydantic.root_validator
+    def _check_xor_input_args(cls, values: dict) -> dict:
+        # Method 1 - connection string
+        connection_string = values.get("connection_string")
+        # Method 2 - individual args (account, user, and password are bare minimum)
+        account = values.get("account")
+        user = values.get("user")
+        password = values.get("password")
+
+        if not bool(connection_string) ^ bool(account and user and password):
+            raise ValueError(
+                "Must provide either a connection string or a combination of account, user, and password."
+            )
+        return values
+
+    @classmethod
+    def _get_exec_engine_excludes(cls) -> set[str]:
+        sql_datasource_fields: set[str] = set(SQLDatasource.__fields__.keys())
+        snowflake_fields: set[str] = set(SnowflakeDatasource.__fields__.keys())
+        return snowflake_fields.difference(sql_datasource_fields)
+
+    def _get_connect_args(self) -> dict[str, str | bool]:
+        excluded_fields: set[str] = set(SQLDatasource.__fields__.keys())
+        # dump as json dict to force serialization of things like AnyUrl
+        return self._json_dict(exclude=excluded_fields, exclude_none=True)
+
+    def get_engine(self) -> sqlalchemy.Engine:
+        if self.connection_string != self._cached_connection_string or not self._engine:
+            try:
+                model_dict = self.dict(
+                    exclude=self._get_exec_engine_excludes(),
+                    config_provider=self._config_provider,
+                )
+
+                kwargs = model_dict.pop("kwargs", {})
+                connection_string = model_dict.pop("connection_string")
+
+                if connection_string:
+                    self._engine = sa.create_engine(connection_string, **kwargs)
+                else:
+                    self._engine = self._build_engine_with_connect_args(**kwargs)
+
+            except Exception as e:
+                # connection_string has passed pydantic validation, but still fails to create a sqlalchemy engine
+                # one possible case is a missing plugin (e.g. psycopg2)
+                raise SQLDatasourceError(
+                    "Unable to create a SQLAlchemy engine from "
+                    f"connection_string: {self.connection_string} due to the "
+                    f"following exception: {str(e)}"
+                ) from e
+            # Since a connection string isn't strictly required for Snowflake, we conditionally cache
+            if self.connection_string:
+                self._cached_connection_string = self.connection_string
+        return self._engine
+
+    def _build_engine_with_connect_args(self, **kwargs) -> URL:
+        connect_args = self._get_connect_args()
+        connect_args.update(kwargs)
+        url = URL(**connect_args)
+        return sa.create_engine(url)
```

Comment:
```suggestion
    def _build_engine_with_connect_args(self, **kwargs) -> sqlalchemy.Engine:
        connect_args = self._get_connect_args()
        connect_args.update(kwargs)
        url = URL(**connect_args)
        return sa.create_engine(url)
```
Does `sa.create_engine()` return an `Engine` or a `URL`?